registry/api/v2/names_test.go

```diff
@@ -149,3 +149,143 @@ func TestRepositoryNameRegexp(t *testing.T) {
 		}
 	}
 }
+
+func TestRepositoryNameRegexp(t *testing.T) {
+	for _, testcase := range []struct {
+		input   string
+		invalid bool
+	}{
+		{
+			input: "short",
+		},
+		{
+			input: "simple/name",
+		},
+		{
+			input: "library/ubuntu",
+		},
+		{
+			input: "docker/stevvooe/app",
+		},
+		{
+			input: "aa/aa/aa/aa/aa/aa/aa/aa/aa/bb/bb/bb/bb/bb/bb",
+		},
+		{
+			input: "aa/aa/bb/bb/bb",
+		},
+		{
+			input: "a/a/a/b/b",
+		},
+		{
+			input:   "a/a/a/a/",
+			invalid: true,
+		},
+		{
+			input:   "a//a/a",
+			invalid: true,
+		},
+		{
+			input: "a",
+		},
+		{
+			input: "a/aa",
+		},
+		{
+			input: "aa/a",
+		},
+		{
+			input: "a/aa/a",
+		},
+		{
+			input:   "foo.com/",
+			invalid: true,
+		},
+		{
+			// currently not allowed by the regex
+			input:   "foo.com:8080/bar",
```

Comment:
@aaronlehmann Just add a TODO here indicating this should be valid when we move the `reference` package.